### PR TITLE
DOC: update np.around docstring with note about floating-point error

### LINKED ISSUE #14391: np.around uses an inexact algorithm, not mentioned in docstring
I came across this from [this discussion](https://news.ycombinator.com/item?id=20818662) of how CPython rounds floating point values to decimal places in `round()`. The two examples below show what CPython does (correct), and the last two what numpy does with a fast but inexact strategy:

```python
  >>> round(56294995342131.5, 2)
  56294995342131.5
  >>> round(56294995342131.5, 3)
  56294995342131.5

  >>> np.round(56294995342131.5, 2)
  56294995342131.5
  >>> np.round(56294995342131.5, 3)
  56294995342131.51
```

I don't see any past issue for this, and the `around` code and docstring seems to have been last touched over 11 years ago, since the last commit is the copy-n-paste from numeric.

The `np.around` docstring does contain the cryptic remark that "Results may also be surprising due
to the inexact representation of decimal fractions in the IEEE floating point standard [1]_ and errors introduced when scaling by powers of ten."

If it is indeed the case that there is no "fast" algorithm for decimal rounding, I think the docstring should at least be elaborated to make clear what approximation is being made. Eg, that we compute `np.true_divide(x*10**d, 10**d)`, which has floating point error, instead of a more accurate BigInt computation.

### PR DESCRIPTION
**Pull request by ahaldane**
Fixes #14391

This just updates the docstring to clarify how this function is inexact for some floating-point values.

(An alternative might be to try to "fix" the np.around algorithm, either by using the BigInt algorithm of the float-printing routines, some simplified bigger-int method, and perhaps also special casing the range of values that are problematic. It's hard to tell how much faster the current algorithm is to a BigInt algorithm, because I find it hard to separate the vectorization speedups in numpy from the round optimizations vs plain python).

### CODE REVIEW

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -3125,10 +3125,35 @@ def around(a, decimals=0, out=None):
...
+        56294995342131.51
+
+    If your goal is to print such values with a fixed number of decimals, it is
+    preferable to use numpy's float printing routines to limit the number of
+    printed decimals:
+
+        >>> np.format_float_positional(56294995342131.5, precision=3)
+        '56294995342131.5'
+
+    The float-printing routines use an accurate but much more computationally
```
Comment (charris) [inline numpy/core/fromnumeric.py]: Remove hyphen in float-printing?

File: numpy/core/fromnumeric.py [hunk 2]
```diff
@@ -3125,10 +3125,35 @@ def around(a, decimals=0, out=None):
...
     rounds to the nearest even value. Thus 1.5 and 2.5 round to 2.0,
-    -0.5 and 0.5 round to 0.0, etc. Results may also be surprising due
-    to the inexact representation of decimal fractions in the IEEE
-    floating point standard [1]_ and errors introduced when scaling
-    by powers of ten.
+    -0.5 and 0.5 round to 0.0, etc.
+
+    `np.around` uses a fast but sometimes inexact algorithm to round
+    floating-point datatypes. For positive `decimals` it is equivalent to
+    `np.true_divide(np.rint(a * 10**decimals), 10**decimals)`, which is inexact
```
Comment (charris) [inline numpy/core/fromnumeric.py]: Double backticks for code.

File: numpy/core/fromnumeric.py [hunk 3]
```diff
@@ -3125,10 +3125,35 @@ def around(a, decimals=0, out=None):
     -----
     For values exactly halfway between rounded decimal values, NumPy
     rounds to the nearest even value. Thus 1.5 and 2.5 round to 2.0,
-    -0.5 and 0.5 round to 0.0, etc. Results may also be surprising due
-    to the inexact representation of decimal fractions in the IEEE
-    floating point standard [1]_ and errors introduced when scaling
-    by powers of ten.
+    -0.5 and 0.5 round to 0.0, etc.
+
+    `np.around` uses a fast but sometimes inexact algorithm to round
```
Comment (charris) [inline numpy/core/fromnumeric.py]: Probably want double backticks here, the single ticks would try to make a link, which seems redundant.

File: numpy/core/fromnumeric.py [hunk 4]
```diff
@@ -3125,10 +3125,35 @@ def around(a, decimals=0, out=None):
...
+    floating-point datatypes. For positive `decimals` it is equivalent to
+    `np.true_divide(np.rint(a * 10**decimals), 10**decimals)`, which is inexact
+    for large floating-point values or large values of `decimal` due the
+    inexact representation of decimal fractions in the IEEE floating point
+    standard [1]_ and errors introduced when scaling by powers of ten. For
+    instance, note the extra "1" in the following:
```
Comment (charris) [inline numpy/core/fromnumeric.py]: Does this need "::'? I'm a bit concerned that some of these might make it as doctests.  Or is that intended? Most doctests are not indented.

File: numpy/core/fromnumeric.py [hunk 5]
```diff
@@ -3125,10 +3125,35 @@ def around(a, decimals=0, out=None):
...
+    printed decimals:
+
+        >>> np.format_float_positional(56294995342131.5, precision=3)
+        '56294995342131.5'
+
+    The float-printing routines use an accurate but much more computationally
+    demanding algorithm to compute the number of digits after the decimal
+    point.
+
+    Alternatively, Python's builtin `round` function uses a more accurate
```
Comment (charris) [inline numpy/core/fromnumeric.py]: I think this link to round is correct. @mattip?

**Comment by eric-wieser** [comment]
> An alternative might be to try to "fix" the np.around algorithm

We should update the docs anyway, then we can leave it in there as a note about how it behaved on older versions if we do eventually fix it.

File: numpy/core/fromnumeric.py (hunk 5 above)
Comment (mattip) [reply numpy/core/fromnumeric.py]: The [rendering of the PR](https://9043-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.around.html#numpy.around) links to the python documentation here

File: numpy/core/fromnumeric.py (hunk 4 above)
Comment (mattip) [reply numpy/core/fromnumeric.py]: It seems to work, see the [rendered doc](https://9043-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.around.html#numpy.around)

File: numpy/core/fromnumeric.py (hunk 3 above)
Comment (mattip) [reply numpy/core/fromnumeric.py]: This and some others in this paragraph appear in italics in the [rendered doc](https://9043-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.around.html#numpy.around), they should have double backticks or be fixed to resolve to a link.

File: numpy/core/fromnumeric.py (hunk 4 above)
Comment (mattip) [reply numpy/core/fromnumeric.py]: nit: decimals is a parameter, so it only really needs single backticks not double. It is find as code as well, so I am not sure it has to be changed.

**Comment by mattip** [comment]
LGTM, one small nit about `decimals`. @charris any more comments?

**Comment by charris** [comment]
@mattip No, I expect you are more familiar with the document markup than I am at this point. You might just go ahead and fix the nit.

**Comment by ahaldane** [comment]
Fixed up the decimals quotes, I think.

**Comment by mattip** [comment]
Thanks @ahaldane it [reads well](https://9065-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.around.html#numpy.around) now.

**Pull request merged**